src/poetry/inspection/info.py

```diff
@@ -369,17 +369,10 @@ def from_setup_files(cls, path: Path) -> PackageInfo:
             name=result.get("name"),
             version=result.get("version"),
             summary=result.get("description", ""),
-            requires_dist=requirements or None,
+            requires_dist=requirements,
```

Comment:
IIRC, this is written to the cache. It shouldn't matter if it's an empty list or None, should it?

Reply:
maybe?  if it doesn't  matter then it doesn't matter: but an empty list seems a clearer representation of what we understand at this point, I don't see why this code would prefer `None`.

perhaps the field shouldn't even be allowed to be `None` on the `PackageInfo`.

Reply:
If i recall correct, the setup reader cannot be trusted when the `setup.py` files are complex. The `None` state was, if i recall correct, used to indicate whether we are confident enough to use the value or resolution must continue (see http/pypi repo code). 

If we are happy to say that we are confident the setup reader has provided the correct value, this should be okay and an empty list is probably more accurate.

It might be that we just need to someone factor in complexity of the setup file to use `[]` or `None`.

Reply:
you probably want to catch up with #9000, which is intended to have arranged the setup reader so that either
- it is confident that it has recognised one of the ways to write setup.py that it does recognise
- or it throws a SetupReaderError (and we do not reach this branch)

Reply:
Awesome, yeah missed that. With that, I think the fix here is good.